### DISCUSSION TITLE
VSC extension toggle style changes

### DISCUSSION THREAD

**Discussion by elijaholmos** [Sysmagine/SemanticDiff#43] [General]
On the website, I noticed that there is a popup to toggle showing the style changes in the diff.

![image](https://github.com/Sysmagine/SemanticDiff/assets/35435704/9f64d5f7-da6d-42bd-8305-a7a398272107)

I wasn't able to find this option in the VS code extension; does it exist?

**Comment by slackner** [comment]
Hello @elijaholmos,

I am sorry if the graphics on our website have caused any misunderstanding. We designed those graphics about a year ago to share our vision, but we focused on implementing the most requested features first.

So far, we haven't gotten a lot of requests to enable/disable style changes, so you're right, it's not yet available in VS Code. Based on your request, I just played around a bit and it should be fairly easy to implement with the framework we already have in place.

However, I'd like to understand your use case a bit better first. How should the feature work exactly? Would you like to reveal the style changes for a specific line range or the whole file?

Best Regards,
Sebastian

**Reply by elijaholmos** [reply]
No worries! Thank you for the explanation. I would imagine this feature revealing the style changes for the entire diff of the file, while I am inside the diff view in vscode

**Reply by slackner** [reply]
Thanks for the answer! What would be the benefit compared to the regular VS code diff (which always shows style changes such as changed whitespace, commas, etc.)?

I assume the main idea is that you want to be able to compare the two, so you probably want to be able to switch back and forth while preserving a specific location in the file, is that correct?

**Reply by elijaholmos** [reply]
Yeah, ideally I would be able to use SemanticDiff as my primary diff viewer and ditch the VSCode one entirely. Another thing that could be hidden/toggled separately are the renaming/refactoring changes, as another use case.
